### PR TITLE
Feature/ci container build publish

### PR DESCRIPTION
**Pull request by j4qfrost**
This still needs some work with tagging, but overall the pipeline works for automatic image builds to the github repo.

### CODE REVIEW

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -0,0 +1,88 @@
+name: Create and publish a Docker image
```
Comment (pwdel) [inline .github/workflows/docker.yml:1]: Perhaps more descriptive might be, "CI/CD Docker Image Build and Publish"

File: .github/workflows/docker.yml [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+          github-token: ${{ secrets.DOCKER_TOKEN }}
+          push: true
+          tags: ${{ steps.backend-meta.outputs.tags }},${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}-backend:latest
+          labels: ${{ steps.backend-meta.outputs.labels }}
+      - name: Generate artifact attestation Backend
+        uses: actions/attest-build-provenance@v1
+        with:
+          subject-name: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME}}-backend
+          subject-digest: ${{ steps.backend.outputs.digest }}
+          push-to-registry: true
```
Comment (pwdel) [inline .github/workflows/docker.yml:88]: Ran through https://rhysd.github.io/actionlint/, no error detected.

**Review by pwdel — APPROVED** [review]

**Review by astrosnat — APPROVED** [review]

**Comment by j4qfrost** [comment]
> LGTM. I'm not familiar with the workflows here, but I'm guessing you don't need to call out the postgres and nginx images directly since we're not actually making any modifications to them that the app is dependent on. @j4qfrost is my understanding of that correct?

Not currently, just focusing on building the app itself. Postgres and nginx changes should be handled by the end user based on their specific infrastructure. We kind of just have them in the compose files for convenience. Benchmarking should be done at some point to see how well the app works with different provisioned resources, but way later in terms of scope.

File: .github/workflows/docker.yml (hunk 1 above)
Comment (j4qfrost) [reply .github/workflows/docker.yml:1]: This is technically just the CI part. I don't think it's too necessary either way.

File: .github/workflows/docker.yml (hunk 2 above)
Comment (j4qfrost) [reply .github/workflows/docker.yml:88]: I'll keep this in mind as a possible later ticket.

**Pull request merged**